airflow/api_connexion/endpoints/variable_endpoint.py

```diff
@@ -14,41 +14,76 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
+from typing import Optional
 
-# TODO(mik-laj): We have to implement it.
-#     Do you want to help? Please look at: https://github.com/apache/airflow/issues/8142
+from flask import Response, make_response, request
+from marshmallow import ValidationError
 
+from airflow.api_connexion.exceptions import BadRequest, NotFound
+from airflow.api_connexion.schemas.variable_schema import variable_collection_schema, variable_schema
+from airflow.models import Variable
+from airflow.utils.session import provide_session
 
-def delete_variable():
+
+def delete_variable(variable_key: str) -> Response:
     """
     Delete variable
     """
-    raise NotImplementedError("Not implemented yet.")
+    Variable.delete(variable_key)
+    return make_response('', 204)
 
 
-def get_variable():
+def get_variable(variable_key: str) -> Response:
     """
     Get a variables by key
     """
-    raise NotImplementedError("Not implemented yet.")
+    try:
+        var = Variable.get(variable_key)
+    except KeyError:
+        raise NotFound("Variable not found")
+    return variable_schema.dump({"key": variable_key, "val": var})
 
 
-def get_variables():
+@provide_session
+def get_variables(session, limit: Optional[int], offset: Optional[int] = None) -> Response:
```

Comment:
```suggestion
def get_variables(session) -> Response:
```
limit and offset are query parameters.

Reply:
query params are passed in as keyword arguments by connexion, see: https://github.com/zalando/connexion/blob/a250e5a59f32ce9d52754f5f7c1827f9abcf0eaf/connexion/decorators/parameter.py#L102